bootstrap/src/basis/stream.mli

```diff
@@ -34,63 +38,76 @@ val init_indef: f:('state -> ('a * 'state) option) -> 'state -> 'a t
 
 (** {1 Length} *)
 
+(* val length: 'a >e t >e-> usize *)
 val length: 'a t -> usize
 (** Return stream length.  This forces and counts every node and is Θ(n). *)
 
+(* val is_empty: 'a >e t >e-> bool  *)
 val is_empty: 'a t -> bool
 (** Return [true] if stream length is 0; [false] otherwise. *)
 
 (** {1 Element access} *)
 
+(* val hd: 'a >e t >e-> 'a *)
 val hd: 'a t -> 'a
 (** Return head (first) stream element, or halt if stream is empty. *)
 
+(* val tl: 'a >e t >e-> 'a >e t *)
 val tl: 'a t -> 'a t
 (** Return tail (a stream of all elements except head), or halt if stream is
     empty. *)
 
 (** {1 Combining and partitioning} *)
 
+(* val push: 'a -> 'a >e t -> 'a >e t *)
 val push: 'a -> 'a t -> 'a t
 (** Push element onto stream and return resulting stream with element as head.
 *)
 
+(* val pop: 'a >e t -> 'a * 'a >e t *)
 val pop: 'a t -> 'a * 'a t
 (** Pop head element off stream and return the decomposed element and remainder
     stream or halt if the stream is empty. *)
 
+(* val concat: 'a >e t -> 'a >e t -> 'a >e t *)
 val concat: 'a t -> 'a t -> 'a t
 (** Concatenate two streams. *)
 
+(* val split: usize -> 'a >e t -> 'a >e t * 'a >e t *)
```

Comment:
`split` and `rev_split` potentially have to force in order to find the split point, right? That means `>e->` for the rightmost arrow. Actually, I think this comment applies to all the following functions.

Reply:
I had to look at the implementations to see what actually happens. No work is actually done until one of the streams is forced.